### PR TITLE
Remove depthshadowauto

### PR DESCRIPTION
**Pull request by GraphineCharles**
This pr is ready to land

This PR is part of a "chain" that needs to land in the folowing order:
- https://ono.unity3d.com/unity/unity/pull-request/127051/_/graphics/remove_depthshadowauto_v3 (landed + backport to 2021.2 also landed)
- https://github.com/Unity-Technologies/Graphics/pull/4509 (this pr)
- https://ono.unity3d.com/unity/unity/pull-request/126494/_/graphics/remove_depthshadowauto_deprecate

The first pr introduces an alternative api to ShadowAuto, DepthAuto and VideoAuto from the GraphicsFormat enum. The second PR to the SRP repo replaces the use of the old api with this newly introduced api . Finally the last pr (https://ono.unity3d.com/unity/unity/pull-request/126494/_/graphics/remove_depthshadowauto_deprecate) effectively deprecates the api. This is landed in two parts to trunk so the Univeral RP can be upgraded before the deprecation lands to avoid deprecation errors there.

---
### Purpose of this PR
- Remove use of the soon to be deprecated AutoShadow and AutoDepth graphics formats.

---
### Testing status
- Ran ABV (fails but it seems not related to this pr)

---

### CODE REVIEW

**Comment by phi-lira** [comment]
Adding @lukaschod and @jonuuukas for awareness.

**Review by CertifiedRice — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]

**Review by lukaschod — APPROVED** [review]

**Review by jonuuukas — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/ShadowUtils.cs [hunk 1]
```diff
@@ -264,10 +260,15 @@ public static void SetupShadowCasterConstantBuffer(CommandBuffer cmd, ref Visibl
 
         public static RenderTexture GetTemporaryShadowTexture(int width, int height, int bits)
         {
-            var shadowTexture = RenderTexture.GetTemporary(width, height, bits, m_ShadowmapFormat);
+
+            var format = Experimental.Rendering.GraphicsFormatUtility.GetDepthStencilFormat(bits, 0);
+            RenderTextureDescriptor rtd = new RenderTextureDescriptor(width, height, Experimental.Rendering.GraphicsFormat.None, format);
```
[the remark this answers is not in the document]
Comment (GraphineCharles) [reply com.unity.render-pipelines.universal/Runtime/ShadowUtils.cs:264]: This is not part of my work, but yes it got recently introduced by a different pr of the texturing team.

**Comment by phi-lira** [comment]
Spoke with @GraphineCharles and the failures are known issues in master already.

**Review by phi-lira — APPROVED** [review]

**Review by simon-engelbrecht-soerensen — APPROVED** [review]
Appears to be working with decals(no regression in test scene), and Boat attack, so approving.

**Pull request merged**